File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        m = defaultdict(int)
        for mt in matches:
            m[mt[0]] = m.get(mt[0], 0)
            m[mt[1]] = m.get(mt[1], 0)+1
        
        zero = []
        one = []
        
        for k,v in m.items():
            if v==0:
                zero.append(k)
            elif v==1:
                one.append(k)
        
        zero.sort()
        one.sort()
        return [zero, one]

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        Map<Integer, Integer> map = new TreeMap<>();
        for(int[] m: matches){
            map.put(m[0], map.getOrDefault(m[0], 0));
            map.put(m[1], map.getOrDefault(m[1], 0)+1);
        }
        
        var zero = new ArrayList<Integer>();
        var one = new ArrayList<Integer>();
        
        for(int k: map.keySet()){
            if(map.get(k)==0) zero.add(k);
            else if (map.get(k)==1) one.add(k);
        }
        
        List<List<Integer>> ans = List.of(zero, one);
        return ans;
    }
}